rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,169 @@
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call.
+
+## Configuration UX
+
+The plugin service will be configured using the existing Teleport YAML file in a section called 'plugin_service'.
+
+```
+plugin_service:
+    enabled: true
+    plugins:
+    - "type": "opsgenie"
+    - "type": "pagerduty"
+    opsgenie:
+        api_key: "path/to/key.txt" # Path to a file containing the Opsgenie API key
+```
+This example would only match plugin resources with the the labels 'type:opsgenie', or 'type:slack'.
+
+The Opsgenie plugin (and any others created) can then be configured using resources. 
+Example plugin.yaml for Opsgenie that would match with this.
+```
+kind: plugin
+metadata:
+  name: opsgenie-plugin
+  labels:
+    type: opsgenie
+spec:
+  opsgenie:
+    addr: "example.app.opsgenie.com" # Address of Opsgenie
+    priority: "2" # Priority to create Opsgenie alerts with
+    alert_tags: ["example-tag"] # List of tags to be added to alerts created in Opsgenie
+    default_schedules: ["schedule1"] # Default on-call schedules to check if none are provided in the access request annotations
+```
+
+Given the above example teleport.yaml the following plugin resource will not match.
+```
+kind: plugin
+metadata:
+  name: slack-plugin
+  labels:
+    type: slack
+spec:
+  slack:
+    addr: "example.slack.com"
+```
+
+The logging configuration will be shared with the main Teleport process.
+
+### Getting an Opsgenie API key
+
+In the Opsgenie web UI go to Settings -> App settings -> API key management. Create key with Read, Create and Update access.
+
+### Executing
+
+The plugin will be started using a command of the form given the appropriate 'plugin_service' is enabled in the config.
+
+```
+teleport start --config /etc/teleport.yaml
+```
+
+## UX
+
+
+### Opsgenie plugin
+Once an access request has been created, the Opsgenie plugin will create an alert in the service specified in the request annotation using the Opsgenie Alert API [Create](https://docs.opsgenie.com/docs/alert-api#create-alert) endpoint. 
+
+The appropriate on-call responder can then click the provided link to the access request and approve or deny it.
+
+For auto approval of certain access requests the access request will be auto approved if the requesting user is on-call in one of the services provided in access request annotations.
+
+Once an access request has been approved or denied the plugin will add a note to the alert and close the relevant alert tied to that access request.
+
+Example role with services in the annotation that indicate that users with this roll can be on-call for those services.
+
+```
+kind: role
+metadata:
+  name: someRole
+spec:
+  allow:
+    request:
+      roles: [someOtherRole]
+      annotations:
+        opsgenie_services: ["service1", "service2"] # These are the Opsgenie services alerts will be created under
```

Comment:
PagerDuty plugin uses 2 annotations: `pagerduty_notify_service` to indicate the service where alert will be created and `pagerduty_services` to indicate the on-call services for auto-approval flow.

You only have `opsgenie_services` defined here currently - which one is that? Can we make it consistent with how PD plugin works?

Also, I'm not very clear on terminology. Below you say that we need to know the "schedule" for the on-call user - so is it a schedule or a service?